src/twisted/trial/_synctest.py

```diff
@@ -1328,7 +1328,8 @@ def mktemp(self):
         )
         if not os.path.exists(base):
             os.makedirs(base)
-        dirname = tempfile.mkdtemp("", "", base)
+        # workaround https://github.com/python/cpython/issues/51574
+        dirname = os.path.relpath(tempfile.mkdtemp("", "", base))
```

Comment:
Maybe add a comment with 3.11

When we will remove support for 3.11 we can do a search in the code and refactor this.

```suggestion
        # This can be removed once we no longer support 3.11
        # Workaround for mkdtemp returning a relative path
        # in 3.11 or older.
        # https://github.com/python/cpython/issues/51574
        dirname = os.path.relpath(tempfile.mkdtemp("", "", base))
```

Reply:
the workaround can't be removed when 3.11 support is removed

Reply:
True. I have updated  the comment.